assembly.py

```diff
@@ -366,46 +378,58 @@ def parser_gapfill_gap2seq(parser=argparse.ArgumentParser(description='Close gap
 
 
 def assemble_spades(
-    inBam,
-    outFasta,
+    in_bam,
+    out_fasta,
     spades_opts='',
-    previously_assembled_contigs=None,
-    mem_limit_gb=4,
+    contigs_trusted=None, contigs_untrusted=None,
+    filter_contigs=False,
+    kmer_sizes=(55,65),
+    mask_errors=False,
```

Comment:
The `trim_rmdup_subsamp_reads` step runs in a reasonable amount of time and it might be worth adding an `n_reads` parameter (perhaps defaulted to `None`) that runs `trim_rmdup_subsamp_reads` (only `if n_reads`) before running SPAdes itself. That would mean the `assemble_spades` entry point could either be utilized completely in place of `assemble_trinity` or it could be used afterwards / in combination, depending on the pipelining code/choices.

I envision that, at first, we might use SPAdes optionally / sometimes, or in combination with. But eventually, it'd be nice to see if it could stand on its own and could replace Trinity. It's true that SPAdes seems able to handle larger inputs, so the "subsamp" part of `trim_rmdup_subsamp_reads` seems less important for SPAdes (or at least, the default value might be a few orders of magnitude higher), but the other steps (trim, rmdup) might be good to keep for consistency's sake at first.

Reply:
Also, I'd give SPAdes singleton reads even if there are enough paired ones, since it knows to treat them separately.

Reply:
True, that makes sense. If that's the case, then I'd lean towards not running any of that preprocessing at all for SPAdes, and in the snakemake rule `assemble_spades` below, it should take in the taxfilt bam, not the subsamp bam.

Reply:
The trim part does makes sense to run before SPAdes though, and rmdup may make sense as well.  So it's just a different version of trim_rmdup_subsamp step, with less subsamp and with a flag to keep all singleton reads.  Maybe also with a trimmomatic flag based on MAXINFO .

Reply:
oh that's easy, just run the existing `trim_rmdup_subsamp_reads` with `n_reads=1000000000` or something really large.  maybe even maxint.